### DISCUSSION TITLE
Shiki

### DISCUSSION THREAD

**Discussion by ap-1** [catppuccin/catppuccin#1839] [Port Requests]
**Are you available to help in the creation of this port? 😄 (optional)**
Maybe

**Additional context 📷 (optional)**
Shiki is notable for being one of astrojs' [default syntax highlighters](https://docs.astro.build/en/reference/api-reference/#built-in-components).

From briefly glancing at their [theme document](https://github.com/shikijs/shiki/blob/main/docs/themes.md), they seem to support TextMate themes and VSCode themes. We can experiment with existing configs in [bat](https://github.com/catppuccin/bat), [sublime-text](https://github.com/catppuccin/sublime-text), and [vscode](https://github.com/catppuccin/vscode).

**Comment by syndrizzle** [comment]
I don't think that we need another theme just for shiki when [Catppuccin for VSCode exists now](https://github.com/catppuccin/vscode)

**Reply by ap-1** [reply]
You're right about that. I don't think Shiki deserves its own repository when the VSCode theme is available, but there's an issue that might come up. People looking for a Shiki theme are likely just going to google something like "catppuccin for shiki," and won't end up finding anything. 

They'd find out about the VSCode theme compatibility if they dig a little deeper, but I think the Catppuccin org could benefit from a single repository that provides instructions for every application/tool that doesn't deserve its own repository. Shiki is a good example of what might fit there. A simple .md file linking to Shiki's theme installation docs and the VSCode theme would be sufficient.

**Reply by syndrizzle** [reply]
Sure

**Reply by yavko** [reply]
Yeah, but the theme is built in code, and shiki uses json themes. How would you even use it with shiki?

**Reply by nekowinston** [reply]
@yavko 
> Yeah, but the theme is built in code, and shiki uses json themes. How would you even use it with shiki?

The VSCode repository hosts the compiled JSON files in a separate branch, [mentioned in the README](https://github.com/catppuccin/vscode#usage):
https://github.com/catppuccin/vscode/tree/compiled

**Reply by yavko** [reply]
> @yavko
> 
> > Yeah, but the theme is built in code, and shiki uses json themes. How would you even use it with shiki?
> 
> The VSCode repository hosts the compiled JSON files in a separate branch, [mentioned in the README](https://github.com/catppuccin/vscode#usage): [catppuccin/vscode@`compiled`](https://github.com/catppuccin/vscode/tree/compiled)

Oh thanks!
